app/scripts/components/share-control/share-control-component.jsx

```diff
@@ -0,0 +1,47 @@
+import React, { Component } from 'react';
+import PropTypes from 'prop-types';
+import classnames from 'classnames';
+
+import Icon from 'components/ui/Icon';
+
+
+class ShareControl extends Component {
+  handleClick(event) {
+    const { open } = this.props;
+    const { href, origin, pathname, search } = window.location;
+    event.preventDefault();
+    this.props.setOpen(!open);
+
+    this.props.setLinks({
+      link: href,
+      embed: `${origin}/embed${pathname}${search}`
+    });
+  }
+
+  render() {
+    const classNames = classnames({
+      [this.props.className]: !!this.props.className
+    });
+
+    return (
+      <div className={`c-share-control ${classNames}`}>
+        <button
+          type="button"
+          className="c-button-map"
+          onClick={e => this.handleClick(e)}
+        >
+          <Icon name="icon-share" className="-small" />
+        </button>
+      </div>
+    );
+  }
+}
+
+ShareControl.propTypes = {
+  setOpen: PropTypes.func,
+  setLinks: PropTypes.func,
+  open: PropTypes.bool,
```

Comment:
This is only a recommendation. Try to order PropTypes by type leaving the functions at the end. It's more readable